src/sage/schemes/elliptic_curves/ell_rational_field.py

```diff
@@ -187,21 +194,23 @@ def __init__(self, ainvs, **kwds):
         EllipticCurve_number_field.__init__(self, Q, ainvs)
 
         if 'conductor' in kwds:
-            self._set_conductor(kwds['conductor'])
+            self._set_conductor(kwds.pop('conductor'))
         if 'cremona_label' in kwds:
-            self._set_cremona_label(kwds['cremona_label'])
+            self._set_cremona_label(kwds.pop('cremona_label'))
         if 'gens' in kwds:
-            self._set_gens(kwds['gens'])
+            self._set_gens(kwds.pop('gens'))
         if 'lmfdb_label' in kwds:
-            self._lmfdb_label = kwds['lmfdb_label']
+            self._lmfdb_label = kwds.pop('lmfdb_label')
         if 'modular_degree' in kwds:
-            self._set_modular_degree(kwds['modular_degree'])
+            self._set_modular_degree(kwds.pop('modular_degree'))
         if 'rank' in kwds:
-            self._set_rank(kwds['rank'])
+            self._set_rank(kwds.pop('rank'))
         if 'regulator' in kwds:
-            self.__regulator = (kwds['regulator'], True)
+            self.__regulator = (kwds.pop('regulator'), True)
         if 'torsion_order' in kwds:
-            self._set_torsion_order(kwds['torsion_order'])
+            self._set_torsion_order(kwds.pop('torsion_order'))
+        if kwds:
+            raise TypeError(f"Unexpected keyword arguments: {kwds}")
 
```

Comment:
Here too.

Reply:
Alright, but actually there's quite a few messages that doesn't follow that convention.

```bash
[schemes]$ grep -r "Error.*'[A-Z]"
affine/affine_homset.py:            raise TypeError('F must be a numerical field')
affine/affine_space.py:            raise ValueError('Must have the same base ring')
elliptic_curves/ell_curve_isogeny.py:        raise ValueError('E1 must be a short Weierstrass curve')
elliptic_curves/ell_curve_isogeny.py:        raise ValueError('E2 must be a short Weierstrass curve')
elliptic_curves/ell_field.py:            raise ValueError('Hasse invariant only defined in positive characteristic')
elliptic_curves/ell_generic.py:            raise ValueError('Frobenius isogeny only exists in positive characteristic')
elliptic_curves/ell_number_field.py:            raise ValueError('There is insufficient data to determine the rank - 2-descent gave lower bound %s and upper bound %s' % (lower, upper))
elliptic_curves/ell_point.py:                raise ValueError('Value %s illegal for point order' % value)
elliptic_curves/ell_point.py:                    raise ValueError('Value %s illegal: outside max Hasse bound' % value)
elliptic_curves/ell_point.py:                raise ValueError('Value %s illegal: %s * %s is not the identity' % (value, value, self))
elliptic_curves/ell_point.py:            raise ValueError('This point %s is not of order n=%s' % (P, n))
elliptic_curves/ell_point.py:            raise ValueError('This point %s is not in Ker(pi - 1)' % P)elliptic_curves/ell_point.py:            raise ValueError('Point %s not in Ker(pi - q)' % Q)
elliptic_curves/ell_point.py:            raise ValueError('ECDLog problem has no solution (order does not divide order of base)')
```

Reply:
Yes.